addons/web_editor/static/src/js/editor/snippets.editor.js

```diff
@@ -1073,6 +1080,32 @@ var SnippetsMenu = Widget.extend({
             }
         });
     },
+    /**
+     * Adds an entry for every invisible snippet in the left panel box.
+     * The entries will contains an 'Edit' button to activate their snippet.
+     *
+     * @private
+     */
+    _updateInvisibleDOM: function () {
+        this.invisibleDOMMap = new Map();
+        $(this.invisibleDOMPanelEl).empty();
+        const $invisibleSnippets = this.getEditableArea().find('.o_snippet_invisible');
+        $invisibleSnippets.removeData('invId');
+
+        _.each($invisibleSnippets, el => {
+            const invId = _.uniqueId('invId');
+            $(el).data('invId', invId);
+            const $invisEntry = $('<p/>', {text: el.getAttribute('string')}).append(
+                $('<a/>', {
+                    href: '#',
+                    text: _t("Edit"),
+                    class: 'ml-2 o_we_invisible_entry',
+                }).data('invId', invId)
+            );
+            $(this.invisibleDOMPanelEl).append($invisEntry);
+            this.invisibleDOMMap.set(invId, [$invisEntry, $(el)]);
```

Comment:
You don't need `invId` anywhere at all. You can just use the <a/> element as map key :+1:

Reply:
I read that using dom element as a map key was not reliable but as discussed it should be ok in ES6